sys/include/shell.h

```diff
@@ -196,6 +197,34 @@ static inline void shell_run(const shell_command_t *commands,
     shell_run_forever(commands, line_buf, len);
 }
 
+/**
+ * @brief   Define shell command
+ *
+ * This macro is a helper for defining a shell command and adding it to the
+ * shell commands XFA (cross file array).
+ *
+ * This requires the module `shell_command_xfa`.
+ *
+ * @experimental This should be considered experimental API, subject to change
+ *               without notice!
+ *
+ * Example:
+ *
+ * ```.c
+ * #include "shell.h"
+ * static int _my_command(int argc, char **argv) {
+ *   // ...
+ * }
+ * SHELL_COMMAND(my_command, "my command help text", _my_command);
+ * ```
+ */
+#if defined(DOXYGEN) || defined(MODULE_SHELL_COMMAND_XFA)
```

Comment:
```suggestion
#if defined(DOXYGEN) || IS_USED(MODULE_SHELL_COMMAND_XFA)
```

We should finally add a coccinelle check for that...